frontend/src/features/admin-form/common/mutations.ts

```diff
@@ -55,6 +64,73 @@ export const useMutateCollaborators = () => {
     [formId, queryClient],
   )
 
+  const getMappedBadRequestErrorMessage = (
+    formCollaboratorAction: FormCollaboratorAction,
+  ) => {
+    let badRequestErrorMessage
+    switch (formCollaboratorAction) {
+      case FormCollaboratorAction.ADD:
+        badRequestErrorMessage = `Please ensure that the email entered is a valid government email. If the error still persists, refresh and try again later.`
+        break
+      default:
+        badRequestErrorMessage = `Sorry, an error occurred. Please refresh the page and try again later.`
+    }
+
+    return badRequestErrorMessage
+  }
+
+  const getMappedDefaultErrorMessage = (
+    formCollaboratorAction: FormCollaboratorAction,
+  ) => {
+    let defaultErrorMessage
+    switch (formCollaboratorAction) {
+      case FormCollaboratorAction.ADD:
+        defaultErrorMessage = 'Error adding collaborator.'
+        break
+      case FormCollaboratorAction.UPDATE:
+        defaultErrorMessage = 'Error updating collaborator.'
+        break
+      case FormCollaboratorAction.REMOVE:
+        defaultErrorMessage = 'Error removing collaborator.'
+        break
+      case FormCollaboratorAction.REMOVE_SELF:
+        defaultErrorMessage = 'Error removing self.'
+        break
+      case FormCollaboratorAction.TRANSFER_OWNERSHIP:
+        defaultErrorMessage = 'Error transfering form ownership.'
+        break
+      //should not reach
+      default:
+        defaultErrorMessage = 'Error.'
+    }
+    return defaultErrorMessage
+  }
+
+  const getMappedErrorMessage = (
+    error: HttpError | Error,
+    formCollaboratorAction: FormCollaboratorAction,
+    requestEmail?: string,
+  ) => {
+    // check if error is an instance of HttpError to be able to access status code of error
+    if (error instanceof HttpError) {
+      let errorMessage
+      switch (error.code) {
+        case 422:
+          requestEmail = typeof requestEmail !== 'undefined' ? requestEmail : ''
+          errorMessage = `${requestEmail} is not part of a whitelisted agency`
```

Comment:
The error message will not be correct if `requestEmail` is an empty string. You probably need a conditional block to format 2 distinct error messages.

If `requestEmail` is this block can in fact never be `undefined` and converted to empty string, let's try to make that explicit in code.